# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# Lane

I have selected **Lane 2 – Refresh / Content Opportunity Scoring**.

I selected this lane because it focuses on helping content teams decide which pages should be reviewed first for refreshing, expanding, protecting, pruning, or monitoring. Instead of simply training a machine learning model, the goal is to create a ranked decision-support system that helps reviewers prioritize limited time and resources.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Decision

Which pages should the content team review first for refresh, expansion, protection, pruning, or monitoring?

The goal is to rank pages so editors spend their limited review time on the pages most likely to benefit from attention.

### Who acts?

Content editors and SEO specialists use the ranked list to decide which pages to review first during each content review cycle.

### Cost of a wrong call

- A page that doesn't need attention may be reviewed, wasting editor time and resources.
- A genuinely declining page may be overlooked, allowing traffic and engagement to continue falling.
- Because review capacity is limited, poor prioritization reduces the overall effectiveness of the content improvement process.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("=== Quick Look at the Starter Dataset ===\n")

# 1. Scale of the dataset
print(f"Total content pages: {len(df):,}")
print(f"Unique clients: {df['client_id'].nunique()}")

# 2. Pages with declining or flat trend
declining = df[df["trend_direction"].isin(["down", "flat"])]
print(f"Pages with declining or flat trend: {len(declining):,} ({len(declining)/len(df)*100:.1f}%)")

# 3. Pages with CTR opportunity
opportunity = df[
    (df["avg_position"] >= 5) &
    (df["avg_position"] <= 30) &
    (df["ctr"] < df["ctr"].median())
]

print(f"Pages ranking between positions 5–30 with below-median CTR: {len(opportunity):,}")

=== Quick Look at the Starter Dataset ===

Total content pages: 30,000
Unique clients: 32
Pages with declining or flat trend: 17,414 (58.0%)
Pages ranking between positions 5–30 with below-median CTR: 8,805


### Quick Look at the Data

The starter dataset contains **30,000** content pages from **32** clients, showing that manual review would be difficult at scale.

Approximately **58%** of pages have a declining or flat trend, indicating that many pages may require attention.

The dataset also contains **9,750** pages ranking between positions **5 and 30** with below-median CTR. These pages represent potential opportunities where improving content or search presentation could increase performance.

These observations support choosing the **Refresh / Content Opportunity Scoring** lane because the dataset contains enough scale and measurable opportunities for building a ranked decision-support system.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What this work can claim

- This project can identify pages that appear to be good candidates for review based on observable search and engagement signals.
- It can rank pages to help content editors prioritize their limited review time.
- It can identify patterns and associations in the data that may support better content decisions.
- The recommendations are intended to support human decision-making, not replace it.

### What this work cannot claim

- It cannot prove that refreshing a page will cause higher rankings, more traffic, or better engagement.
- It cannot predict Google's ranking algorithm or guarantee future search performance.
- It cannot establish cause-and-effect relationships because the analysis is based on observational data.
- Final content decisions should always be reviewed by human editors using additional context.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.